emmaemb/vizualization.py

```
@@ -94,16 +93,36 @@ def plot_emb_space(
     else:
         raise ValueError(f"Method {method} not implemented")
 
-    fig = px.scatter(
-        x=embeddings_2d[:, 0],
-        y=embeddings_2d[:, 1],
-        color=emma.metadata[color_by] if color_by else None,
-        labels={"color": color_by},
-        title=f"{emb_space} embeddings after {method}",
-        hover_data={"Sample": emma.sample_names},
-        opacity=0.5,
-        color_discrete_map=(emma.color_map[color_by] if color_by else None),
-    )
+    # args for px.scatter
+    scatter_args = {
+        "x": embeddings_2d[:, 0],
+        "y": embeddings_2d[:, 1],
+        "labels": {"color": color_by},
+        "title": f"{emb_space} embeddings after {method}",
+        "hover_data": {"Sample": emma.sample_names},
+        "opacity": 0.5,
+    }
+
+    # categorical column
+    try:
+        emma._check_column_is_categorical(color_by)
```

Can we account for color_by == None here?

---

We actually are🙈 `emma._check_column_is_categorical` raises a `ValueError` when the column is not categorical (like `None` for example), then we jump to `except: pass` below, i.e. doing nothing.

Just noticed that I do still have `"labels": {"color": color_by},` above in the initial `scatter_args` though, I'll remove that since we don't need it.

---

Great - thanks for clarifying!